crates/storage/codecs/src/alloy/transaction/optimism.rs

```diff
@@ -0,0 +1,61 @@
+use crate::Compact;
+use alloy_primitives::{Address, Bytes, TxKind, B256, U256};
+use op_alloy_consensus::TxDeposit as AlloyTxDeposit;
+use reth_codecs_derive::add_arbitrary_tests;
+use serde::{Deserialize, Serialize};
+
+/// Deposit transactions, also known as deposits are initiated on L1, and executed on L2.
+///
+/// This is a helper type to use derive on it instead of manually managing `bitfield`.
+///
+/// By deriving `Compact` here, any future changes or enhancements to the `Compact` derive
+/// will automatically apply to this type.
+///
+/// Notice: Make sure this struct is 1:1 with [`op_alloy_consensus::TxDeposit`]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Default, Serialize, Deserialize, Compact)]
+#[cfg_attr(test, derive(arbitrary::Arbitrary))]
+#[add_arbitrary_tests(compact)]
+pub(crate) struct TxDeposit {
+    source_hash: B256,
+    from: Address,
+    to: TxKind,
+    mint: Option<u128>,
+    value: U256,
+    gas_limit: u64,
+    is_system_transaction: bool,
+    input: Bytes,
+}
+
+impl Compact for AlloyTxDeposit {
+    fn to_compact<B>(&self, buf: &mut B) -> usize
+    where
+        B: bytes::BufMut + AsMut<[u8]>,
+    {
+        let tx = TxDeposit {
+            source_hash: self.source_hash,
+            from: self.from,
+            to: self.to,
+            mint: self.mint,
+            value: self.value,
+            gas_limit: self.gas_limit as u64,
+            is_system_transaction: self.is_system_transaction,
+            input: self.input.clone(),
```

Comment:
@mattsse currently all `Compact` impls for alloy transactions clone input during encoding. should we address this with a helper struct?

Reply:
I think this is fine because this is `Bytes` and should be cheap enough